Develop

haduigon opened the pull request:
https://haduigon.github.io/js_2048_game/

loralevitska reviewed (changes requested):
hi, good job!  need some improvements

- change text 
![image](https://github.com/mate-academy/js_2048_game/assets/70009205/73d42dea-c98a-4850-b9f7-574bc2d491be)

![image](https://github.com/mate-academy/js_2048_game/assets/70009205/cdbd4c9e-076e-43e6-b130-6fcb6bd81e7a)
- fix number appearing, without delay

haduigon commented:
> hi, good job! need some improvements
> 
> * change text
>   ![image](https://private-user-images.githubusercontent.com/70009205/313472926-73d42dea-c98a-4850-b9f7-574bc2d491be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODExMDYsIm5iZiI6MTcxMDY4MDgwNiwicGF0aCI6Ii83MDAwOTIwNS8zMTM0NzI5MjYtNzNkNDJkZWEtYzk4YS00ODUwLWI5ZjctNTc0YmMyZDQ5MWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDEzMDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZWFlN2MwNWFmNjY1NTQ2ZjhjYjg0NjA4NzEzNjdjMTE1ZWFkNGY3MDZkZWNkZGJmNzM2N2U4MjQwYTUzNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1vPH4Gk21CKVfyxNkCI6-ZY1TlVpfOXiGttJYCrYKho)
> 
> ![image](https://private-user-images.githubusercontent.com/70009205/313472859-cdbd4c9e-076e-43e6-b130-6fcb6bd81e7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODExMDYsIm5iZiI6MTcxMDY4MDgwNiwicGF0aCI6Ii83MDAwOTIwNS8zMTM0NzI4NTktY2RiZDRjOWUtMDc2ZS00M2U2LWIxMzAtNmZjYjZiZDgxZTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDEzMDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExMjFiOTBiZmM2MTViMDgzNTZiODJjMjQ4MjYzMWM1ZTM1YzNhZjMxNmY0NzkyMWU3OWI4NGI0YzBjYmI1ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.64z8Vp0jmN2wE86q1m8xcf8qS8VkIozhzXjlEArX5XA)
> 
> * fix number appearing, without delay

game over message appears if there are no available moves, screenshot is attached. Or should I change text for "Game over" ?

<img width="1792" alt="Screenshot 2024-03-17 at 15 09 29" src="https://github.com/mate-academy/js_2048_game/assets/20277989/3fdc5c7e-0e49-49d8-a3e7-b822747c9f7c">

haduigon commented:
> hi, good job! need some improvements
> 
> * change text
>   ![image](https://private-user-images.githubusercontent.com/70009205/313472926-73d42dea-c98a-4850-b9f7-574bc2d491be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODExMDYsIm5iZiI6MTcxMDY4MDgwNiwicGF0aCI6Ii83MDAwOTIwNS8zMTM0NzI5MjYtNzNkNDJkZWEtYzk4YS00ODUwLWI5ZjctNTc0YmMyZDQ5MWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDEzMDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZWFlN2MwNWFmNjY1NTQ2ZjhjYjg0NjA4NzEzNjdjMTE1ZWFkNGY3MDZkZWNkZGJmNzM2N2U4MjQwYTUzNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1vPH4Gk21CKVfyxNkCI6-ZY1TlVpfOXiGttJYCrYKho)
> 
> ![image](https://private-user-images.githubusercontent.com/70009205/313472859-cdbd4c9e-076e-43e6-b130-6fcb6bd81e7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODExMDYsIm5iZiI6MTcxMDY4MDgwNiwicGF0aCI6Ii83MDAwOTIwNS8zMTM0NzI4NTktY2RiZDRjOWUtMDc2ZS00M2U2LWIxMzAtNmZjYjZiZDgxZTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDEzMDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExMjFiOTBiZmM2MTViMDgzNTZiODJjMjQ4MjYzMWM1ZTM1YzNhZjMxNmY0NzkyMWU3OWI4NGI0YzBjYmI1ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.64z8Vp0jmN2wE86q1m8xcf8qS8VkIozhzXjlEArX5XA)
> 
> * fix number appearing, without delay

I deleted the delay before new number appears

anastasiiavorobiova reviewed (changes requested):
Great work! Consider changing the start button to restart after the beginning of the game. New values shouldn't be added if the game field doesn't change

haduigon commented:
> Great work! Consider changing the start button to restart after the beginning of the game. New values shouldn't be added if the game field doesn't change

it works ) after first step not after pushing start button, start is changing to restart, screen is attached
<img width="1792" alt="Screenshot 2024-03-17 at 15 28 37" src="https://github.com/mate-academy/js_2048_game/assets/20277989/4987fd0f-f094-4786-8624-37ba482b9a92">

anastasiiavorobiova reviewed (changes requested):
Consider using the example as a reference https://play2048.co/ If you feel some of the comments are unclear, please ask for help in the chat or come to the Q&A session. We will deal with everything)

haduigon commented:
> Consider using the example as a reference https://play2048.co/ If you feel some of the comments are unclear, please ask for help in the chat or come to the Q&A session. We will deal with everything)

I have considered this example a lot of times ) please, give me strict instructions what exactly I should correct ? I realized everything what was described in the task with the messages which were in the task) the mark up you provided defers from task mark up

VitaliyBondarenko1982 reviewed (changes requested):
Almost done.
Markup doesn't matter.
There is two bugs or mismatches which was considered in previous comments and not fixed:
![chrome-capture-2024-3-17](https://github.com/mate-academy/js_2048_game/assets/33029864/d360405d-b68c-44d8-8a46-c6506cc49649)
- after click on start button - button should change state on replace (text and color). before first move. now it happening after first move.
- when no cells for merge and no any cell moved - no need to add new cell. on gif I always press arrow on top. and when was such state of board (see screenshot) - new cell shouldn't appear. but now it appears.
![Screenshot from 2024-03-17 15-47-47](https://github.com/mate-academy/js_2048_game/assets/33029864/58ac0365-330c-4537-bdbc-0ef45f1cb4a3)

haduigon commented:
> Almost done. Markup doesn't matter. There is two bugs or mismatches which was considered in previous comments and not fixed: ![chrome-capture-2024-3-17](https://private-user-images.githubusercontent.com/33029864/313477383-d360405d-b68c-44d8-8a46-c6506cc49649.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU2NjgsIm5iZiI6MTcxMDY4NTM2OCwicGF0aCI6Ii8zMzAyOTg2NC8zMTM0NzczODMtZDM2MDQwNWQtYjY4Yy00NGQ4LThhNDYtYzY1MDZjYzQ5NjQ5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE0MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNGM2NWQxMTdkOWRiMDY1ZmUyOTg0YWRkYTg3OWY5NTA1OGMxMmEwMmM0Nzk1YTgwYTVhNDdjYmUyNjUwNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hZ-PSfpCROLatBEmXkBKJQKYWxRScPi-8C6vfL9_R3I) [ ![chrome-capture-2024-3-17](https://private-user-images.githubusercontent.com/33029864/313477383-d360405d-b68c-44d8-8a46-c6506cc49649.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU2NjgsIm5iZiI6MTcxMDY4NTM2OCwicGF0aCI6Ii8zMzAyOTg2NC8zMTM0NzczODMtZDM2MDQwNWQtYjY4Yy00NGQ4LThhNDYtYzY1MDZjYzQ5NjQ5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE0MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNGM2NWQxMTdkOWRiMDY1ZmUyOTg0YWRkYTg3OWY5NTA1OGMxMmEwMmM0Nzk1YTgwYTVhNDdjYmUyNjUwNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hZ-PSfpCROLatBEmXkBKJQKYWxRScPi-8C6vfL9_R3I) ](https://private-user-images.githubusercontent.com/33029864/313477383-d360405d-b68c-44d8-8a46-c6506cc49649.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU2NjgsIm5iZiI6MTcxMDY4NTM2OCwicGF0aCI6Ii8zMzAyOTg2NC8zMTM0NzczODMtZDM2MDQwNWQtYjY4Yy00NGQ4LThhNDYtYzY1MDZjYzQ5NjQ5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE0MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNGM2NWQxMTdkOWRiMDY1ZmUyOTg0YWRkYTg3OWY5NTA1OGMxMmEwMmM0Nzk1YTgwYTVhNDdjYmUyNjUwNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hZ-PSfpCROLatBEmXkBKJQKYWxRScPi-8C6vfL9_R3I) [ ](https://private-user-images.githubusercontent.com/33029864/313477383-d360405d-b68c-44d8-8a46-c6506cc49649.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU2NjgsIm5iZiI6MTcxMDY4NTM2OCwicGF0aCI6Ii8zMzAyOTg2NC8zMTM0NzczODMtZDM2MDQwNWQtYjY4Yy00NGQ4LThhNDYtYzY1MDZjYzQ5NjQ5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE0MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNGM2NWQxMTdkOWRiMDY1ZmUyOTg0YWRkYTg3OWY5NTA1OGMxMmEwMmM0Nzk1YTgwYTVhNDdjYmUyNjUwNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hZ-PSfpCROLatBEmXkBKJQKYWxRScPi-8C6vfL9_R3I)
> 
> * after click on start button - button should change state on replace (text and color). before first move. now it happening after first move.
> * when no cells for merge and no any cell moved - no need to add new cell. on gif I always press arrow on top. and when was such state of board (see screenshot) - new cell shouldn't appear. but now it appears.
>   ![Screenshot from 2024-03-17 15-47-47](https://private-user-images.githubusercontent.com/33029864/313477638-58ac0365-330c-4537-bdbc-0ef45f1cb4a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU2NjgsIm5iZiI6MTcxMDY4NTM2OCwicGF0aCI6Ii8zMzAyOTg2NC8zMTM0Nzc2MzgtNThhYzAzNjUtMzMwYy00NTM3LWJkYmMtMGVmNDVmMWNiNGEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE0MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxMDdlNzIyNTIwZjlmYzdlOGFhMjFkMzNlZDNkZmZjMTQ3OTcyN2QxYzQ1MmQ4OTA3ZDk2NzFjOGRhYmE4MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-lVFLqZ8HGPGQedGDRexPY2xSj7bUvaBrIbHWoQWNCs)

both are done )

maxim2310 reviewed (changes requested):
Good work, but lets make some improvement 

1. i must be able click on reset anytime 
2. on restart scores not resets
3. can appear only one block 
![image](https://github.com/mate-academy/js_2048_game/assets/96265844/e880e12d-bae3-4666-a1f0-7c11e4aac702)
4. and you give to much scores for merge (2 + 2 = 4 but you give 8)
5. i am not sure about it)
![image](https://github.com/mate-academy/js_2048_game/assets/96265844/2386c463-9150-4f8a-8a8e-1c59c6a8c50f)
6. on restart winer block must be hidden

haduigon commented:
> Good work, but lets make some improvement
> 
> 1. i must be able click on reset anytime

It is mentioned in the task, that player can restart game only after first move ) I can change it after the only pushing the start button, but want to be sure we are doing right )

<img width="1742" alt="screenForPullRequest" src="https://github.com/mate-academy/js_2048_game/assets/20277989/900c9e0c-1a74-42a5-8352-8d0102394ea8">


> 2. on restart scores not resets
done
> 3. can appear only one block
>    ![image](https://private-user-images.githubusercontent.com/96265844/313794473-e880e12d-bae3-4666-a1f0-7c11e4aac702.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NDc1MjMsIm5iZiI6MTcxMDg0NzIyMywicGF0aCI6Ii85NjI2NTg0NC8zMTM3OTQ0NzMtZTg4MGUxMmQtYmFlMy00NjY2LWExZjAtN2MxMWU0YWFjNzAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDExMjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiMTY2NDM5N2RhMzMyNjFkOGYyN2ZkNDRkYzRmZGZlYWQzNGM5N2U1MWQzMDFkYTZjZGQ1ZWFkMWM4N2NlYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1NFUQB99jKNX5w4j27dn2sbkhstQ0VjfQxFHbDyh1cI)

cant reproduce the mistake
> 4. and you give to much scores for merge (2 + 2 = 4 but you give 8)

fixed
> 5. i am not sure about it)
>    ![image](https://private-user-images.githubusercontent.com/96265844/313795271-2386c463-9150-4f8a-8a8e-1c59c6a8c50f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NDc1MjMsIm5iZiI6MTcxMDg0NzIyMywicGF0aCI6Ii85NjI2NTg0NC8zMTM3OTUyNzEtMjM4NmM0NjMtOTE1MC00ZjhhLThhOGUtMWM1OWM2YThjNTBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDExMjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlZDhiYTdiYjYxMTc2Zjg3OTI3YWZhNTZhNWY1NTA1NmU2N2RjZjcyNTYxNWE2NjcxMzQ4NjFmY2M4YzI2OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DPMv0T-W8Llg_zuVaidLb0u_UMXeZCfr9FJBDHirmlk)

fixed)
> 6. on restart winer block must be hidden

fixed )
